packages/perseus-editor/src/hint-editor.tsx

```diff
@@ -363,40 +363,29 @@ class CombinedHintsEditor extends React.Component<CombinedHintsEditorProps> {
         ]);
         this.props.onChange({hints: hints}, () => {
             const i = hints.length - 1;
-            // eslint-disable-next-line react/no-string-refs
-            // @ts-expect-error - TS2339 - Property 'focus' does not exist on type 'ReactInstance'.
-            this.refs["hintEditor" + i].focus();
+            this.hintEditors[i]?.focus();
         });
     };
 
     getSaveWarnings: () => any = () => {
-        return _.chain(this.props.hints)
+        return this.props.hints
             .map((hint, i) => {
-                return _.map(
-                    // eslint-disable-next-line react/no-string-refs
-                    // @ts-expect-error - TS2339 - Property 'getSaveWarnings' does not exist on type 'ReactInstance'.
-                    this.refs["hintEditor" + i].getSaveWarnings(),
-                    (issue) => "Hint " + (i + 1) + ": " + issue,
-                );
+                return this.hintEditors["hintEditor" + i]
+                    ?.getSaveWarnings()
+                    .map((issue) => "Hint " + (i + 1) + ": " + issue);
             })
-            .flatten(true)
-            .value();
+            .flat();
     };
 
-    serialize: (options?: any) => ReadonlyArray<string> = (options: any) => {
+    serialize(options?: any): ReadonlyArray<PerseusRenderer> {
```

Comment:
Refactoring `serialize` to be a method instead of an arrow function changes semantics - the value of `this` might be unbound.

Reply:
This `serialize` is called directly by the `EditorPage` in its `serialize()` function so it is safe. As with `serializeHint`, I think it's useful to "undo" these migrations to arrow functions so that it's more obvious we aren't passing these around by value.